oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthenticationCallbackHandler.java

```diff
@@ -512,6 +512,17 @@ public APIGatewayProxyResponseEvent handleRequest(
                                 > 0) {
                     userSession.setCurrentCredentialStrength(lowestRequestedCredentialTrustLevel);
                 }
+                var currentCredentialStrength =
+                        userInfo.getStringClaim(
+                                AuthUserInfoClaims.CURRENT_CREDENTIAL_STRENGTH.getValue());
+                if (isNull(currentCredentialStrength)
+                        || lowestRequestedCredentialTrustLevel.compareTo(
+                                        CredentialTrustLevel.valueOf(currentCredentialStrength))
+                                > 0) {
+                    orchSessionService.updateSession(
+                            orchSession.withCurrentCredentialStrength(
+                                    lowestRequestedCredentialTrustLevel));
```

Comment:
This saves earlier than sharedSession, but in the same `try` block. I'm pretty sure that means our sharedSession and orchSession could go out of sync if the code between fails (eg if `generateAndSaveAuthorisationCode` fails). I think best to have the save be the last action, since we only want to save the new state if everything succeeds.

Reply:
I think I do agree with what you were saying but also hesitant as with this save ordering I was trying to keep to the old code as much as possible. In case ordering was important due to failures.